apps/src/face_detection/face_trainer.cpp

```diff
@@ -25,10 +25,9 @@ main(int argc, char** argv)
   pcl::console::parse_argument(argc, argv, "-use_normals", use_normals);
   pcl::console::parse_argument(argc, argv, "-num_images", num_images);
 
-  pcl::RFFaceDetectorTrainer fdrf;
```

Comment:
There are some value like forest_fn which are used in parse_argument and in the setters of struct. Shall i remove them and delete the corresponding parse_argument lines.
My understanding so far is not to remove them.

Reply:
No, please don't remove them. Users should have the option to specify custom values for these variables.
I am not sure why you added `{}` behind `fdrf`? What does that change?
And please don't delete the call to `setWSize`. Even if 80 is currently the default value in the `RFFaceDetectorTrainer` constructor, it is better to set that explicitly here (could change in the constructor, and setting the `int` barely costs anything)

Reply:
As far as, i know it has some impact in case of min_votes_size_for example. Only with {}  it is default initialized. I do not see it being used whenever a new struct is used and it could have some side effects.


Example: https://godbolt.org/z/rz3Y4MjY4 i is randomly initialized in first case while in the second case it is default initialized

Reply:
Ah, ok. So I would say the actual problem is that `min_votes_size_` is not initialized when the object is constructed. It should be initialized in the initialization list or inline (as you did for the other members of that class). Relying on the braced-init-list to initialize it to zero is not transparent. There could be two options: init it to a default value, or init it to a dummy value and check whether this dummy value has been overwritten via `setFaceMinVotes` before using the value. The second option would be the option of choice if there is no sensible default value. For `min_votes_size_`, I would say it makes sense to init it to zero (explicitly, in the initialization list or inline). This would be equivalent to having no min votes size (as default).